src/Microsoft.Azure.WebJobs.Host/Bindings/AttributeCloner.cs

```diff
@@ -107,15 +103,46 @@ private BindingDataResolver GetResolver(PropertyInfo propInfo, INameResolver nam
                 return GetAppSettingResolver((string)originalValue, appSettingAttr, nameResolver, propInfo);
             }
             // try to resolve with auto resolve ({...}, %...%)
-            if (autoResolveAttr != null && originalValue != null)
+            if (autoResolveAttr != null)
             {
-                _autoResolves[propInfo] = autoResolveAttr;
-                return GetTemplateResolver((string)originalValue, autoResolveAttr, nameResolver, propInfo, contract);
+                if (propInfo.PropertyType != typeof(string))
+                {
+                    throw new InvalidOperationException($"Property '{propInfo.Name}' with AutoResolve must be type string.");
+                }
+                var str = (string)originalValue;
+
+                if (string.IsNullOrWhiteSpace(str))
+                {
+                    if (autoResolveAttr.Default != null)
```

Comment:
Agree with @mathewc - I'd love to see a refactoring pass over this to try to minimize the if statements here